Participants optimisations

hristoterezov opened the pull request:
<!--
Thank you for your pull request. Please provide a thorough description below.

Contributors guide: https://github.com/jitsi/jitsi-meet/blob/master/CONTRIBUTING.md
-->
We need to squash the commits!

damencho commented on react/features/participants-pane/components/MeetingParticipantList.js:
```diff
@@ -105,14 +105,28 @@ export function MeetingParticipantList() {
+    // It seems that useTranslation is not very scallable. Unmount 500 components that have the useTranslation hook is
+    // taking more than 10s. To workaround the issue we need to pass the texts as props. This is temporary and dirty
+    // solution!!!
+    // One potential proper fix would be to use
```
Can you finish this note :)

damencho commented on react/features/base/participants/reducer.js:
```diff
@@ -62,42 +73,241 @@ const PARTICIPANT_PROPS_TO_OMIT_WHEN_UPDATE = [
+            if (state.everyoneIsModerator && !isModerator) {
+                state.everyoneIsModerator = false;
+            } else if (!state.everyoneIsModerator && isModerator) {
+                state.everyoneIsModerator = _isEveryoneModerator(state);
```
Cause this is the change: we have X participants and there is at least one that is not moderator, an update comes that one participant has become a moderator and we check, `are all now moderators?`

vp8x8 replied:
Yeah, makes sense. Did not think this through.

vp8x8 commented:
This looks great! A few comments:

This is for the case when **moderation is enabled.**
1. When trying to unmute or dismiss a participant who raised its hand it does not work as it should for the moderator. 
2. After a participant was asked to be unmuted, the moderator will still see _Ask to unmute_ button in the participant list.
3. Sometimes the app reloads when a participant leaves and the participant list is open.

I'll try to test more scenarios soon, but these are the most obvious atm.

hristoterezov commented on react/features/base/participants/reducer.js:
```diff
@@ -62,42 +73,241 @@ const PARTICIPANT_PROPS_TO_OMIT_WHEN_UPDATE = [
+
+        state.remote.set(participant.id, participant);
+
+        if (participant.isFakeParticipant) {
```
Currently we store the fake participant in `remote` and in `fakeParticipants` too. The idea is to optimise some loops over the fake participants. But they need to be added in `remote` in order to be rendered. Potentially we can try to not add them in `remote` but this would complicate the rest of the code a lot. What I'm trying to say is that even if I move it upper I won't be able to return early.

hristoterezov commented on react/features/filmstrip/actions.web.js:
```diff
@@ -127,7 +127,8 @@ export function setHorizontalViewDimensions() {
     return (dispatch: Function, getState: Function) => {
-        const participants = getState()['features/base/participants'];
+        const state = getState();
+        const { participants } = [ getLocalParticipant(state), ...getRemoteParticipants(state).values() ];
```
Good catch!

hristoterezov commented:
@vp8x8 Thank you very much for the review! I tested the PR myself a little bit more and fixed some bugs. Also I believe I addressed all your comments except the moderation enabled use case. I'll leave @damencho to handle those.

vp8x8 commented:
This LGTM.
There are a few more problems with moderation but @damencho explained that we can move forward and solve them afterwards.
